pkgs/by-name/xf/xfs_undelete/package.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, coreutils
+, tcl-8_6
+, tcllib
+}:
+
+stdenv.mkDerivation {
+  pname = "xfs_undelete";
+  version = "unstable-2023-04-12";
+
+  src =  fetchFromGitHub {
+    repo = "xfs_undelete";
+    owner = "ianka";
+    rev = "9e2f7abf0d3a466328e335d251c567ce4194e473";
+    sha256 = "0n1718bmr2lfpwx57hajancda51fyrgyk9rbybbadvd8gypvzmhh";
+  };
+
+  buildInputs = [ tcl-8_6 tcllib coreutils ];
+  nativeBuildInputs = [ makeWrapper tcl-8_6.tclPackageHook ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 xfs_undelete -t $out/bin
+    runHook postInstall
+  '';
+
+  fixupPhase = ''
+    runHook preFixup
+    wrapProgram $out/bin/xfs_undelete \
+      --prefix PATH : ${lib.makeBinPath [ tcl-8_6 ]}
+    runHook postFixup
+  '';
```

Comment:
The package seems to get double wrapped. I'm assuming `tclPackageHook` does something similar already? Have you tried running the executable without this wrapper?

Reply:
@h7x4, the `xfs_undelete` binary runs fine, as per my original version of `fixupPhase`.

Reply:
The wrapping used to be necessary on `release-23.11`.  I can check again.

Reply:
It runs fine, but it's a bit wasteful to double wrap it. `$out/bin` currently looks like this:

```
[nix-shell:~/.cache/nixpkgs-review/pr-281881-1]$ ls -la results/xfs_undelete/bin/
total 108
dr-xr-xr-x 1 root root   170 Jan  1  1970 .
dr-xr-xr-x 1 root root    28 Jan  1  1970 ..
-r-xr-xr-x 2 root root 36552 Jan  1  1970 xfs_undelete
-r-xr-xr-x 2 root root 36552 Jan  1  1970 .xfs_undelete-wrapped
-r-xr-xr-x 2 root root 16072 Jan  1  1970 .xfs_undelete-wrapped_
-r-xr-xr-x 4 root root 18811 Jan  1  1970 ..xfs_undelete-wrapped-wrapped
```

[`tclPackageHook` seems to allow you to specify extra args for the wrapper through its environment variable](https://github.com/NixOS/nixpkgs/blob/33dd4c5bc577ded23b94de0c9417cebb0ae5063e/pkgs/development/interpreters/tcl/tcl-package-hook.sh#L5), so we could probably avoid the double wrapping and still have the same effect by replacing the entire `fixupPhase` with this:

```nix
  tclWrapperArgs = [ "--prefix" "PATH" ":" (lib.makeBinPath [ tcl-8_6 ]) ];
```

That should prefix both `PATH` and `TCLLIBPATH` in the same wrapper.

Reply:
Thank you for the suggestion, done!